### PR TITLE
Plus Components

### PR DESCRIPTION
**Pull request by mariamm-alaa**

### CODE REVIEW

File: components/PostForm.vue [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            v-model="localBody"
+            class="mt-1 block input-field"
+            rows="5"
+          ></textarea>
+        </div>
+  
+        <!-- Submit Button -->
+        <button type="submit" class="submit-button">
+          Save Post
+        </button>
```
Comment (A-Elshorafa) [inline components/PostForm.vue:47]: Button

File: components/PostForm.vue [hunk 2]
```diff
@@ -0,0 +1,141 @@
+<template>
+    <div class="form-container">
+      <h1>Create a New Post</h1>
+      <form @submit.prevent="handleSubmit" class="form-content">
```
Comment (A-Elshorafa) [inline components/PostForm.vue:4]: Form

File: components/PostForm.vue [hunk 3]
```diff
@@ -0,0 +1,141 @@
+<template>
+    <div class="form-container">
+      <h1>Create a New Post</h1>
```
Comment (A-Elshorafa) [inline components/PostForm.vue:3]: Label with a text-size

File: components/PostForm.vue [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        <div class="mb-4">
+          <Label for="content" class="block text-sm font-medium text-gray-700">
+            Content
+          </Label>
+          <textarea
+            id="content"
+            v-model="localBody"
+            class="mt-1 block input-field"
+            rows="5"
+          ></textarea>
```
Comment (A-Elshorafa) [inline components/PostForm.vue:41]: use TextArea

File: components/PostForm.vue [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+        <div class="mb-4">
+          <Label for="title" class="block text-sm font-medium text-gray-700">
+            Title
+          </Label>
+          <input
+            type="text"
+            id="title"
+            v-model="localTitle"
+            class="mt-1 block input-field"
+          />
```
Comment (A-Elshorafa) [inline components/PostForm.vue:28]: use their input 
https://www.shadcn-vue.com/docs/components/input.html

File: pages/index.vue [hunk 6]
```diff
@@ -2,16 +2,7 @@
   <div :class="toggleState ? 'dark-theme' : 'light-theme'">
     <h1 class="center-title">All Posts</h1>
```
Comment (A-Elshorafa) [inline pages/index.vue:3]: to label with a text-size

File: components/PostForm.vue (hunk 1 above)
Comment (mariamm-alaa) [reply components/PostForm.vue:47]: done

File: components/PostForm.vue (hunk 2 above)
Comment (mariamm-alaa) [reply components/PostForm.vue:4]: thats already a separate file for the form component

File: components/PostForm.vue (hunk 3 above)
Comment (mariamm-alaa) [reply components/PostForm.vue:3]: done

File: components/PostForm.vue (hunk 4 above)
Comment (mariamm-alaa) [reply components/PostForm.vue:41]: done

File: components/PostForm.vue (hunk 5 above)
Comment (mariamm-alaa) [reply components/PostForm.vue:28]: done

File: pages/index.vue (hunk 6 above)
Comment (mariamm-alaa) [reply pages/index.vue:3]: done

**Pull request merged**